notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb

```diff
@@ -0,0 +1,1557 @@
+    "from models.imagebind_model import ModalityType\n",
+    "\n",
+    "modalities = [ModalityType.TEXT, ModalityType.VISION, ModalityType.AUDIO]\n",
+    "fp_model_paths = {modality: Path(f\"image-bind-{modality}\") / f\"image-bind-{modality}.xml\" for modality in modalities}\n",
+    "int8_model_paths = {modality: Path(f\"image-bind-{modality}\") / f\"image-bind-{modality}_int8.xml\" for modality in modalities}"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "f261e6a4",
+   "metadata": {},
+   "source": [
+    "### Prepare datasets"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "65d8d559",
+   "metadata": {},
+   "source": [
+    "The [LAION-400M](https://laion.ai/blog/laion-400-open-dataset/) dataset contains CLIP-filtered 400 million image-text pairs is used to quantize image and text models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "40d71747",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import data\n",
+    "import os\n",
+    "import requests\n",
+    "import tempfile\n",
+    "\n",
+    "from requests.packages.urllib3.exceptions import InsecureRequestWarning\n",
+    "requests.packages.urllib3.disable_warnings(InsecureRequestWarning)\n",
+    "\n",
+    "def check_text_data(data):\n",
+    "    if isinstance(data, str):\n",
+    "        return True\n",
+    "    if isinstance(data, list):\n",
+    "        return all(isinstance(x, str) for x in data)\n",
+    "    return False\n",
+    "\n",
+    "def laion_preprocess(example, image_column=\"url\", text_column=\"caption\"):\n",
+    "    if not check_text_data(example[text_column]):\n",
+    "        raise ValueError(\"Text data is not valid\")\n",
+    "\n",
+    "    with tempfile.TemporaryDirectory() as tempdir:\n",
+    "        url = example[image_column]\n",
+    "        f_name = os.path.join(tempdir, 'image.jpg')\n",
+    "        try:\n",
+    "            response = requests.get(url, verify=False, timeout=20)\n",
+    "            f_name.write(response.content)\n",
+    "        except Exception:\n",
+    "            return None\n",
+    "\n",
+    "        example[\"pixel_values\"] = data.load_and_transform_vision_data([f_name], \"cpu\")\n",
+    "\n",
+    "    example[\"text\"] = data.load_and_transform_text([example[text_column]], \"cpu\")\n",
+    "    return example\n",
+    "\n",
+    "def collate_fn(examples):\n",
+    "    examples = [laion_preprocess(example) for example in examples]\n",
+    "    assert len(examples) == 1\n",
+    "    if examples[0] is None:\n",
+    "        return None\n",
+    "\n",
+    "    return {\n",
+    "        \"pixel_values\": examples[0][\"pixel_values\"],\n",
+    "        \"input_ids\": examples[0][\"text\"]\n",
+    "    }"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "befc2204",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import load_dataset\n",
```

Comment:
I get an error in this cell (on Windows): 

_ModuleNotFoundError                       Traceback (most recent call last)
Cell In[4], line 1
----> 1 from datasets import load_dataset
      2 import itertools
      3 import torch

ModuleNotFoundError: No module named 'datasets'_

Adding !pip install datasets directly to the notebook helped.

Reply:
Also quantization of the last model failed with
ImportError: To support decoding audio files, please install 'librosa' and 'soundfile'.